### PR TITLE
[Sprint 42] XD-2666 Upgrade to Spring Integration Kafka 1.0.0.RC1

### PR DESCRIPTION
**Pull request by mbogoevici**
- library upgrade
- adjustments to the API
- removed local implementations

### CODE REVIEW

File: modules/source/kafka/config/kafka.xml [hunk 1]
```diff
@@ -68,23 +58,42 @@
 	</bean>
 
 	<bean id="kafkaInboundChannelAdapter"
```
Comment (artembilan) [inline modules/source/kafka/config/kafka.xml]: How about to change to the namespace component - `<int-kafka:message-driven-channel-adapter>` ?

File: extensions/spring-xd-extension-kafka/src/main/java/org/springframework/integration/x/kafka/AutoOffsetResetStrategy.java [hunk 2]
```diff
@@ -14,26 +14,27 @@
...
- * To be refactored once https://jira.spring.io/browse/INTEXT-131 is fixed
+ * The possible values for automatically resetting the offset of Kafka client
  *
  * @author Marius Bogoevici
  */
-public interface DeletableOffsetManager extends OffsetManager {
+public enum AutoOffsetResetStrategy {
+
+	smallest(OffsetRequest.EarliestTime()),
+	largest(OffsetRequest.LatestTime());
```
Comment (artembilan) [inline extensions/spring-xd-extension-kafka/src/main/java/org/springframework/integration/x/kafka/AutoOffsetResetStrategy.java]: Can't they be in upper case?

Comment (mbogoevici) [inline extensions/spring-xd-extension-kafka/src/main/java/org/springframework/integration/x/kafka/AutoOffsetResetStrategy.java]: No, they parse automatically from the module options.

File: modules/source/kafka/config/kafka.xml (hunk 1 above)
Comment (mbogoevici) [inline modules/source/kafka/config/kafka.xml]: I'd rather make that a later enhancement (seeing that all this is internal) and keep current changes to a minimum,

**Comment by artembilan** [comment]
So, nothing to complain about, if all existing tests pass :smile:

**Comment by markpollack** [comment]
squashed, rebased and merged as 044a242d172cb664a87172bee7f306a049bf4c13

**Pull request closed without merging**